src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -66,17 +66,34 @@ const DataProcessingPage = ({ experimentId, experimentData, route }) => {
   ];
 
   const dispatch = useDispatch();
-  const [stepIdx, setStepIdx] = useState(0);
 
   const completedPath = '/experiments/[experimentId]/data-exploration';
 
-  const completedSteps = useSelector((state) => state.experimentSettings.processing.meta.stepsDone);
+  const completedSteps = useSelector((state) => state.experimentSettings.processing.processingConfig.stepsDone);
+
+  const [stepIdx, setStepIdx] = useState((completedSteps?.size ?? 0) % steps.length);
```

Comment:
Why have to modulo (%)? can `completedSteps.size` be greater than `steps.length`?

Reply:
completedSteps.size can be equal to steps.length, while the highest value it should be able to take is steps.length - 1

Reply:
Oh I see..

Reply:
it was to have the latest step be the one that is opened as soon as we arrive to the page, but there is an issue with this that has been fixed in the newest version by adding an `initialState` value and a `useEffect`.